spec/outputs/datadog_metrics_spec.rb

```diff
@@ -1 +1,83 @@
 require "logstash/devutils/rspec/spec_helper"
+require "logstash/outputs/datadog_metrics"
+
+describe LogStash::Outputs::DatadogMetrics do
+  let(:http_instance) { double('net_http_instance') }
+  let(:request) { double('net_http_request') }
+  let(:response) { double('net_http_response') }
+
+  let(:json_obj) { double('json_obj') }
+  let(:logger) { double('logger') }
+  let(:ddtags) { ['foo', 'bar'] }
+  let(:event_hash) do
+    {
+      'message' => 'Hello world!',
+      'source' => 'datadog-metrics',
+      'type' => 'generator',
+      'host' => 'localhost',
+      '@timestamp' => LogStash::Timestamp.now
+    }
+  end
+  let(:event)  { LogStash::Event.new(event_hash) }
+  let(:params) do
+    {
+      'api_key' => '123456789',
+      'dd_tags' => ddtags,
+      'metric_type' => 'gauge'
+    }
+  end
+
+  before do
+    allow(Net::HTTP).to receive(:new).and_return(http_instance)
+    allow(Net::HTTP::Post).to receive(:new).and_return(request)
+  end
+
+  subject { described_class.new(params) }
+
+  before do
+    expect(http_instance).to receive(:use_ssl=)
```

Comment:
running these expectations on the before block seems out of place, if it's something worth testing then it should go to an `it`, otherwise we can remove it?